src/php/Compiler/Analyzer/TypeAnalyzer/SpecialForm/LetSymbol.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phel\Compiler\Analyzer\TypeAnalyzer\SpecialForm;
+
+use Phel\Compiler\Analyzer\AnalyzerInterface;
+use Phel\Compiler\Analyzer\Ast\BindingNode;
+use Phel\Compiler\Analyzer\Ast\LetNode;
+use Phel\Compiler\Analyzer\Environment\NodeEnvironmentInterface;
+use Phel\Compiler\Analyzer\Exceptions\AnalyzerException;
+use Phel\Compiler\Analyzer\TypeAnalyzer\SpecialForm\Binding\DeconstructorInterface;
+use Phel\Lang\Collections\LinkedList\PersistentListInterface;
+use Phel\Lang\Collections\Vector\PersistentVectorInterface;
+use Phel\Lang\Symbol;
+use Phel\Lang\TypeFactory;
+
+final class LetSymbol implements SpecialFormAnalyzerInterface
+{
+    private AnalyzerInterface $analyzer;
+    private DeconstructorInterface $deconstructor;
+
+    public function __construct(AnalyzerInterface $analyzer, DeconstructorInterface $deconstructor)
+    {
+        $this->analyzer = $analyzer;
+        $this->deconstructor = $deconstructor;
+    }
+
+    public function analyze(PersistentListInterface $list, NodeEnvironmentInterface $env): LetNode
+    {
+        if (!($list->get(0) instanceof Symbol && $list->get(0)->getName() === Symbol::NAME_LET)) {
+            throw AnalyzerException::withLocation("This is not a 'let.", $list);
+        }
+
+        if (count($list) < 2) {
+            throw AnalyzerException::withLocation("At least two arguments are required for 'let", $list);
+        }
+
+        if (!($list->get(1) instanceof PersistentVectorInterface)) {
+            throw AnalyzerException::withLocation('Binding parameter must be a vector', $list);
+        }
+
+        if (!(count($list->get(1)) % 2 === 0)) {
+            throw AnalyzerException::withLocation('Bindings must be a even number of parameters', $list);
+        }
+
+        $bindings = $this->deconstructor->deconstruct($list->get(1));
+        $bindingData = [];
+        foreach ($bindings as $binding) {
+            $bindingData[] = $binding[0];
+            $bindingData[] = $binding[1];
+        }
+
+        $newListData = $list->toArray();
+        $newListData[1] = TypeFactory::getInstance()->persistentVectorFromArray($bindingData);
+        $newList = TypeFactory::getInstance()
+            ->persistentListFromArray($newListData)
+            ->copyLocationFrom($list)
+            ->withMeta($list->getMeta());
+
+        return $this->analyzeLetOrLoop($newList, $env);
+    }
+
+    private function analyzeLetOrLoop(PersistentListInterface $list, NodeEnvironmentInterface $env): LetNode
+    {
+        $exprs = $list->rest()->rest()->toArray();
```

Comment:
Minor observation: could we place this variable closer to its usage? I see it's being used only in line 92. Could we move this line 66 down to the 89, for example? Unless there is a reason why is this variable here, the first line in the function, I feel it's too alone and a bit "far away" from its usage...

Reply:
Will fix this.